inductiva/benchmarks/methods.py

```diff
@@ -0,0 +1,351 @@
+            simulator (Simulator): Simulator to run.
+            append (bool): If True, appends the tasks to a benchmark regardless 
+            if that benchmark is new or not. If False, will prompt the user if
+            the benchmark already exists.
+            machine_class (BaseMachineGroup): Class to use to create the machine
+                group. If not given, MachineGroup will be used.
+            input_args (Dict[str, Union[Callable, Any]]): Arguments for the
+                input files.
+            machine_args (Dict[str, Union[Callable, Any]]): Arguments for the
+                machines.
+            simulator_args (Dict[str, Union[Callable, Any]]): Arguments for the
+                simulator.
+    """
+    project = Project(name, append=True)
+    project.open()
+
+    tasks = project.get_tasks()
+
+    if len(tasks) > 0 and not append:
+        print(__("benchmark-already-exists", name))
+        response = input()
+        if response in ("no", "n"):
+            print(__("operation_canceled"))
+            project.close()
+            return
+        elif response in ("yes", "ye", "y"):
+            print(__("appending_tasks"))
+
+    # create dict with resource type: list of tasks ran on that resource
+    # Example
+    # { "n1-standard-4": [task1, task2], "n1-standard-8": [task3]}
+    current_project_tasks = _tasks_by_vm_type(project)
+    to_run = _compute_tasks_to_run(machines, current_project_tasks, replicas)
+
+    print()
+    if not to_run:
+        print(__("all_tasks_run"))
+        project.close()
+        return
+
+    _print_info(current_project_tasks, to_run)
+
+    if input_args is not None:
+        template_manager = TemplateManager(template_dir=input_files)
+        base_path = template_manager.get_root_dir()
+    for machine, current_replicas in to_run.items():
+
+        machine_args_current = _replace_callable_from_dict(
+            machine_args, machine)
+
+        print(f"\nCreating machine group {machine} with {machine_args_current}")
+
+        # Use MachineGroup as default value for machine_class
+        machine_class = machine_class or MachineGroup
+
+        resource = machine_class(machine_type=machine, **machine_args_current)
+
+        simulator_args_current = _replace_callable_from_dict(
+            simulator_args, resource)
+        input_args_current = _replace_callable_from_dict(input_args, resource)
+
+        if input_args_current:
+
+            template_manager.set_root_dir(f"{base_path}/{name}")
+            input_files = template_manager.render_dir(**input_args_current)
+
+        # input_files either come from the template manager or
+        # are passed directly
+        simulator_args_current["input_dir"] = input_files
+        simulator_args_current["on"] = resource
+
+        print(f"\nRunning {simulator}"
+              f"with {simulator_args_current}")
+        if resource is not None:
+            while not _can_start_resource(resource):
```

Comment:
I think we could replace this function with something like:

```
try:
    resource.start()
except ApiException:
   ...
```

Reply:
I rather have the check instead of using exceptions. We might even create a method like inductiva.users.can_run(resource) in the future.